robot/robot.py

```diff
@@ -38,23 +38,26 @@ def __init__(self, robotd_path=ROBOTD_ADDRESS):
             self.known_gamestates,
         ]
 
+        self._display_boards()
         self._wait_for_power_board()
 
+        print('Waiting for start button.')
+        self.power_board.wait_start()
+
+        self.power_board.power_on()
+        print("Starting user code.")
+
+    def _display_boards(self):
+        print("Found the following hardware devices:")
+        for board_list in self.all_known_boards:
+            for board in board_list:
+                print(str(board))  # Force string representation
```

Comment:
Isn't `str(x)` the default behaviour for `print(x)`? (genuinely curious as I'd assumed it was but I don't think I've actually seen any docs on it)

Reply:
Something tells me it uses `__repr__` by default here, or something weird like that. Either way, better explicit

Reply:
Given the surprise factor of this being explicit, I actually think this is better implicit. My reasoning is that, as it stands, this leads to readers questioning what would otherwise happen.  (I'd draw similarities with checking for `None` where it wouldn't reasonably be expected, etc.)

Anyway, it look like `str` _is_ the default behaviour:
``` python
>>> class Foo:
...  def __str__(self):
...   return 'Foo.__str__'
...
>>> repr(Foo())
'<__main__.Foo object at 0x000001AAF0379780>'
>>> str(Foo())
'Foo.__str__'
>>> print(Foo())
Foo.__str__
```

Recall however that the default `__str__` calls `__repr__`.